lib/God.js

```diff
@@ -86,6 +87,21 @@ God.executeApp = function executeApp(env, cb) {
   if (!env_copy.created_at)
     env_copy['created_at'] = Date.now();
 
+  var current_user = '';
+
+  if (os.userInfo)
+    try {
+      current_user = os.userInfo().username;
+    } catch (err) {
+      // For the case of unhandled error for uv_os_get_passwd
+      // https://github.com/Unitech/pm2/issues/3184
+      current_user = '';
+    } else {
+    current_user = process.env.USER || process.env.LNAME || process.env.USERNAME || process.env.SUDO_USER || process.env.C9_USER || process.env.LOGNAME;
+  }
+
+  env_copy['username'] = current_user;
```

Comment:
This part is executed on the daemon side so the current user will always use the one of the daemon.